packages/@lwc/engine-server/src/renderer.ts

```diff
@@ -330,23 +338,114 @@ function isConnected(node: HostNode) {
 // synthetic shadow.
 const insertStylesheet = noop as (content: string, target: any) => void;
 
-// Noop on SSR.
-const addEventListener = noop as (
-    target: HostNode,
+function addEventListener(
+    node: N,
     type: string,
     callback: EventListener,
-    options?: AddEventListenerOptions | boolean
-) => void;
+    useCaptureOrOptions?: AddEventListenerOptions | boolean
+): void {
+    if (node.type !== 'element') {
+        return;
+    }
+
+    let savedCallback = callback;
+
+    if (useCaptureOrOptions) {
+        if (useCaptureOrOptions === true || useCaptureOrOptions.capture) {
```

Comment:
[The spec](https://dom.spec.whatwg.org/#ref-for-dom-eventtarget-addeventlistener%E2%91%A2) is not clear on this, but browsers also allow truthy objects like `1` and `"foo"` here as well. This gets weird because the empty object `{}`, however, is treated as an actual property bag (i.e. `capture: false` rather than `capture: true`).

There is some background discussion on this in https://github.com/WICG/EventListenerOptions/issues/21#issuecomment-190242970 and [this Chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=543685).

I was a bit unsure what behavior they landed on, so I just wrote [a small test](https://bl.ocks.org/nolanlawson/raw/2f38242207bd62ddd8d4da627cfc1503/). Luckily all 3 browser engines behave the same. Basically `useCapture` is true in these cases:

- `true`
- `1`
- `"foo"` (non-empty string)
- `{ capture: true }`

It's false in all the other cases I tested (including truthy values like `{}` and `[]`).

Reply:
This is super helpful, thank you! I'll add these conditions w/ comments.